frontend/include/chpl/uast/Select.h

```diff
@@ -121,6 +121,10 @@ class Select final : public AstNode {
     auto end = begin + numWhenStmts_;
     return AstListIteratorPair<When>(begin, end);
   }
+
+  bool hasOtherwise() const {
+    return whenStmt(numWhenStmts()-1)->isOtherwise();
+  }
```

Comment:
Can a when statement be empty, and if so, would this seg fault?

Reply:
I don't think so, but I don't know the AST construction code well enough to say confidently. Select statements do have to have at least one `when` though.